data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkConfig.java

```diff
@@ -142,8 +145,8 @@ public ObjectKeyOptions getObjectKeyOptions() {
         return objectKeyOptions;
     }
 
-    public PredefinedObjectMetadata getPredefinedObjectMetadata() {
-        return predefinedObjectMetadata;
+    public ObjectMetadata getObjectMetadata() {
```

Comment:
I think you need to return either the predefined or the new one here.

Also, add an `@AssertTrue` to be sure they are not both set by the user.